app/policies/waiver_submission_policy.rb

```diff
@@ -5,6 +5,11 @@ class WaiverSubmissionPolicy < ApplicationPolicy
 
   def launch?
     !!user
```

Comment:
this line doesn't do anything right now, i think you want
```suggestion
    return false if user.nil?
```
The `!!user` with no return only works if it's the last line in the function and you take advantage of Ruby's implicit return.

Reply:
Ah okay, got it, just edited that